form-guidelines.rst

```diff
@@ -0,0 +1,186 @@
+******************
+Form Guidelines
```

Comment:
please retitle.

1. XForm, not just form. 
2. Guidelines? What does that mean? Are these suggestions? Best practices? Things that only matter when XForm is used in Collect? 

It is difficult to know at a glance how this relates to the Form Design page in the other PR.